node/core/provisioner/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+				ToJob::Provisioner(RequestBlockAuthorshipData(_, sender)) => self.provisionable_data_channels.push(sender),
+				ToJob::Provisioner(ProvisionableData(data)) => {
+					for channel in self.provisionable_data_channels.iter_mut() {
+						// REVIEW: the try operator here breaks the run loop if any receiver ever unexpectedly
+						// closes their channel. Is that desired?
+						channel.send(data.clone()).await?;
+					}
+					self.note_provisionable_data(data);
+				}
+				ToJob::Stop => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	// REVIEW: not async, but small; hopefully permissable
+	fn note_provisionable_data(&mut self, provisionable_data: ProvisionableData) {
+		use ProvisionableData::{Bitfield, BackedCandidate};
+
+		match provisionable_data {
+			Bitfield(_, signed_bitfield) => self.signed_bitfields.push(signed_bitfield),
+			BackedCandidate(backed_candidate) => self.backed_candidates.push(backed_candidate),
+			_ => {}
+		}
+	}
+
+	// The provisioner is the subsystem best suited to choosing which specific
+	// backed candidates and availability bitfields should be assembled into the
+	// block. To engage this functionality, a
+	// `ProvisionerMessage::RequestInherentData` is sent; the response is a set of
+	// non-conflicting candidates and the appropriate bitfields. Non-conflicting
+	// means that there are never two distinct parachain candidates included for
+	// the same parachain and that new parachain candidates cannot be included
+	// until the previous one either gets declared available or expired.
+	//
+	// The main complication here is going to be around handling
+	// occupied-core-assumptions. We might have candidates that are only
+	// includable when some bitfields are included. And we might have candidates
+	// that are not includable when certain bitfields are included.
+	//
+	// When we're choosing bitfields to include, the rule should be simple:
+	// maximize availability. So basically, include all bitfields. And then
+	// choose a coherent set of candidates along with that.
+	async fn send_inherent_data(&mut self, sender: oneshot::Sender<ProvisionerInherentData>) -> Result<(), Error> {
+
+		// coherency rules for backed candidates, recap:
+		//
+		// - [x] only one per parachain
+		// - [ ] candidate is available (2/3+ of validators have set the 1 bit)
+		// - [ ] not expired (how can we tell?)
+
+		let availability_cores = self.get_availability_cores().await?;
+		let mut coherent_candidates = Vec::new();
+		let mut parachains_represented = HashSet::new();
+		for candidate in self.backed_candidates.iter() {
+			// only allow the first candidate per parachain
+			// this sequence reliance is _entirely arbitrary_
+			if !parachains_represented.insert(candidate.candidate.descriptor.para_id) {
+				continue
+			}
+
+			// figure out which availability core corresponds to this candidate, then
+			// update its `availability` bitfield with our list of bitfields to include.
+			// then, if it's < 2/3 full, don't include it.
```

Comment:
so, per my comment above, the logic represented by this comment actually is not needed